### PR TITLE
Move People Index Page to the GOV.UK Design System

### PR DESCRIPTION
**Pull request by davidgisbey**
## Description

This PR moves the People index page to the GOV.UK Design System.

1. Duplicates the people controller specs and assign the `Preview design system` permission to the non-legacy specs user
2. Ensures the controller renders the correct layout and view based on the whether the user has the  `Preview design system` permission
2.  Adds the view in the GOV.UK Design System
3. Updates any specs that have broken

## Screenshot

image

image

## Trello card

https://trello.com/c/7BWS3tx1/36-people-list

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

**Comment by nnagewad** [comment]
@davidgisbey looks good. Some quick feedback.

- The text in the [Warning text component](https://design-system.service.gov.uk/components/warning-text/) --> mind using the default [GOVUK text color](https://design-system.service.gov.uk/styles/colour/).
- Just out of curiosity, what happens when the user selects the 'Create person' button?

**Comment by davidgisbey** [comment]
Sure thing. As a rule of thumb should we always be that going forward?

It takes you to the new person page 

image

**Comment by nnagewad** [comment]
> 

I think moving forward align as much with the component sans customization.

As for the button, methinks it should be a link, since it's directing the user to a new page. Happy to get your thoughts bud

**Comment by davidgisbey** [comment]
The Publishing Component or the GOV.UK one? The Publishing Components gem gives us more options so i think we're tending to use those to get it as close to like for like as possible as that's the steer the teams been given.

**Comment by nnagewad** [comment]
> The Publishing Component or the GOV.UK one? The Publishing Components gem gives us more options so i think we're tending to use those to get it as close to like for like as possible as that's the steer the teams been given.

Right on. Had no idea there was a variation [with red font](https://components.publishing.service.gov.uk/component-guide/warning_text/with_red_font).

File: app/views/admin/people/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+} %>
+
+<%= render "govuk_publishing_components/components/button", {
+  text: "Create person",
+  href: new_admin_person_path,
+  secondary_solid: true,
+  margin_bottom: 8
+} %>
+
+<div class="app-c-govuk-table--filterable">
```
[the remark this answers is not in the document]
Comment (DilwoarH) [reply app/views/admin/people/index.html.erb:16]: This change should probably sit in the commit that comes after this one as it explains the context (https://github.com/alphagov/whitehall/pull/7405/commits/0c01cb1a66e6e261aee278d7d43fb61ebde3752f)

[the remark this answers is not in the document]
Comment (davidgisbey) [reply app/views/admin/people/index.html.erb:16]: have moved in into the next commit 👍

**Review by DilwoarH — APPROVED** [review]

**Pull request merged**